tests/integration/preprocessor/_io/test_zarr.py

```diff
@@ -0,0 +1,247 @@
+    zarr_xr = xr.open_dataset(
+        zarr_path,
+        consolidated=True,
+        decode_times=time_coder,
+        engine="zarr",
+        backend_kwargs={},
+    )
+
+    conversion_func = ncdata.iris_xarray.cubes_from_xarray
+    cubes = conversion_func(zarr_xr)
+
+    assert isinstance(cubes, iris.cube.CubeList)
+    assert len(cubes) == 24
+
+
+def test_load_zarr3():
+    """
+    Test loading a Zarr3 store from a https Object Store.
+
+    We have a permanent bucket: esmvaltool-zarr at CEDA's object store
+    "url": "https://uor-aces-o.s3-ext.jc.rl.ac.uk",
+    where will host a number of test files like this one.
+    """
+    zarr_path = (
+        "https://uor-aces-o.s3-ext.jc.rl.ac.uk/"
+        "esmvaltool-zarr/example_field_0.zarr3"
+    )
+
+    # with "dummy" storage options
+    cubes = load(
+        zarr_path,
+        ignore_warnings=None,
+        backend_kwargs={"storage_options": {}},
+    )
+
+    assert len(cubes) == 1
+    cube = cubes[0]
+    assert cube.var_name == "q"
+    assert cube.standard_name == "specific_humidity"
+    assert cube.long_name is None
+    assert cube.units == cf_units.Unit("1")
+    coords = cube.coords()
+    coord_names = [coord.standard_name for coord in coords]
+    assert "longitude" in coord_names
+    assert "latitude" in coord_names
+
+
+def test_load_zarr3_cmip6_via_ncdata():
+    """
+    Test loading a Zarr3 store from a https Object Store.
+
+    This test is meant to determine how much memory we need via the
+    two main API routes we need to go to load an Iris cube from a Zarr
+    store, using an object storage unit:
+
+    - API1: Xarray.open_dataset
+    - API2: ncdata.iris_xarray.cubes_from_xarray
+
+    We have a permanent bucket: esmvaltool-zarr at CEDA's object store
+    "url": "https://uor-aces-o.s3-ext.jc.rl.ac.uk",
+    where will host a number of test files like this one.
+
+    This is an actual CMIP6 dataset (Zarr built from netCDF4 via Xarray)
+    - Zarr store on disk: 243 MiB
+    - compression: Blosc
+    - Dimensions: (lat: 128, lon: 256, time: 2352, axis_nbounds: 2)
+    - chunking: time-slices; netCDF4.Dataset.chunking() = [1, 128, 256]
+
+    Test takes 8-9s (median: 8.5s) and needs max Res mem: 1GB
+    """
+    zarr_path = (
+        "https://uor-aces-o.s3-ext.jc.rl.ac.uk/"
+        "esmvaltool-zarr/pr_Amon_CNRM-ESM2-1_02Kpd-11_r1i1p2f2_gr_200601-220112.zarr3"
+    )
+
+    time_coder = xr.coders.CFDatetimeCoder(use_cftime=True)
+    zarr_xr = xr.open_dataset(
+        zarr_path,
+        consolidated=True,
+        decode_times=time_coder,
+        engine="zarr",
+        backend_kwargs={},
+    )
+    # API1: 420MB memory; 1.5s
+
+    conversion_func = ncdata.iris_xarray.cubes_from_xarray
+    cubes = conversion_func(zarr_xr)
+    # API2: 1GB memory; 8.5s
+
+    assert isinstance(cubes, iris.cube.CubeList)
+    assert len(cubes) == 1
+    assert cubes[0].has_lazy_data()
+
+
+def test_load_zarr3_cmip6_metadata():
```

Comment:
This should be moved to the other tests in `test_load` (or the other way round, but they should not be separated).